src/main.rs

```diff
@@ -10,6 +10,46 @@ use libazureinit::{
 
 const VERSION: &str = env!("CARGO_PKG_VERSION");
 
+fn get_username(
+    imds_body: String,
+) -> Result<String, Box<dyn std::error::Error>> {
+    let username;
+    let password;
+
+    if match imds::get_provision_with_password(&imds_body) {
+        Ok(disable_passwd_auth) => disable_passwd_auth,
+        Err(_err) => {
+            return Err("Failed to get disable password authentication".into())
+        }
+    } {
```

Comment:
You could propagate the error here with `?` rather than matching, although the function being called only returns `Ok` at the moment (or panics). If you want to adjust the error type, you can achieve the same thing by using `map_err` like this:

```suggestion
    if imds::get_provision_with_password(&imds_body).map_err(|_err| {
        "Failed to get disable password authentication".to_string()
    })? {
```

This is totally a style nit since it achieves the same thing, but these feel a bit more "Rust-y" to me.

Maybe not something to change in this PR, but the function name is also confusing since `get_provision_with_password() -> true` when `disablePasswordAuthentication == true` and I would expect the opposite based on the name.

Reply:
`map_err` sounds like a good idea.

And you are right. I have been also confused by the naming all over the place, which is actually the opposite of the actual meaning. But I did not want to touch too many things. ;-)

Reply:
I would go ahead and change the function name to is_password_authentication_disabled

Reply:
Done, changed to `map_err`.

Reply:
renamed the function name to `is_password_authentication_disabled` as well.